python-sdk/src/astro/databases/mssql.py

```diff
@@ -0,0 +1,359 @@
+                "Please use  `sql` param instead.",
+                DeprecationWarning,
+                stacklevel=2,
+            )
+            sql = kwargs.get("sql_statement")  # type: ignore
+
+        # We need to set autocommit=True for specific queries
+        if isinstance(sql, str):
+            if (autocommit is True) or any(
+                query in sql.lower() for query in self._queries_requiring_autocommit
+            ):
+                self.connection.execute(sqlalchemy.text(sql).execution_options(autocommit=True), parameters)
+            else:
+                result = self.connection.execute(
+                    sqlalchemy.text(sql).execution_options(autocommit=autocommit), parameters
+                )
+                return result
+        else:
+            # this is used for append
+            result = self.connection.execute(sql, parameters)
+            return result
+
+    def create_schema_if_needed(self, schema: str | None) -> None:
+        """
+        This function checks if the expected schema exists in the database. If the schema does not exist,
+        it will attempt to create it.
+        :param schema: DB Schema - a namespace that contains named objects like (tables, functions, etc)
+        """
+        if schema and not self.schema_exists(schema):
+            statement = self._create_schema_statement.format(schema)
+            self.run_sql(statement, autocommit=True)
+
+    def create_table_from_select_statement(
+        self,
+        statement: str,
+        target_table: BaseTable,
+        parameters: dict | None = None,
+    ) -> None:
+        """
+        Export the result rows of a query statement into another table.
+        :param statement: SQL query statement
+        :param target_table: Destination table where results will be recorded.
+        :param parameters: (Optional) parameters to be used to render the SQL query
+        """
+
+        statement = self._create_table_statement.format(
+            target_table.metadata.schema,
+            target_table.name,
+            target_table.metadata.database,
+            statement,
+        )
+
+        self.run_sql(statement, parameters, autocommit=True)
+
+    def fetch_all_rows(self, table: BaseTable, row_limit: int = -1) -> list:
+        """
+        Fetches all rows for a table and returns as a list. This is needed because some
+        databases have different cursors that require different methods to fetch rows
+
+        :param row_limit: Limit the number of rows returned, by default return all rows.
+        :param table: The table metadata needed to fetch the rows
+        :return: a list of rows
+        """
+        statement = f"SELECT * FROM {self.get_table_qualified_name(table)}"
+        if row_limit > -1:
+            statement = f"SELECT TOP {row_limit} * FROM {self.get_table_qualified_name(table)}"
+        response = self.run_sql(statement)
+        return response.fetchall()  # type: ignore
```

Comment:
@sunank200  It will result in below error otherwise. This  `# type: ignore` is present in `BaseDatabase` class also. If we want this to be fixed, then we need to do this across the repo.

```src/astro/databases/mssql.py: note: In member "fetch_all_rows" of class "MssqlDatabase":
src/astro/databases/mssql.py:237: error: Returning Any from function declared
to return "List[Any]"  [no-any-return]
            return response.fetchall()
```